test/steps/servicepage.js

```diff
@@ -0,0 +1,41 @@
+/* @flow */
+
+"use strict";
+
+import assert from "assert";
+import Yadda from 'yadda';
+import _ from 'underscore';
+import { By } from 'selenium-webdriver';
+
+import dictionary from "../support/dictionary";
+import unpromisify from "../support/yadda-promise";
+
+async function asyncFilter(arr, check) {
+    var results = arr.map(check);
+    results = await Promise.all(results);
+    return _.zip(results, arr)
+        .filter((e) => e[0])
+        .map((e) => e[1]);
```

Comment:
I think I've addressed everything else, but I don't see a good way to make this work more cleanly.

`results` would be a list of [element, Promise<boolean>] if I'm understanding correctly, so you couldn't use Promise.all with it